lib/color_spinor_field.cpp

```diff
@@ -776,9 +776,22 @@ namespace quda {
                                                    QudaMemoryType new_mem_type) {
     ColorSpinorParam coarseParam(*this);
     for (int d=0; d<nDim; d++) coarseParam.x[d] = x[d]/geoBlockSize[d];
-    coarseParam.nSpin = (nSpin == 1) ? 2 : (nSpin / spinBlockSize); // coarsening staggered check
 
-    coarseParam.nColor = Nvec;
+    // check for optimized-KD build by seeing if total geoBlockSize == 1
```

Comment:
This looks like it will prevent coarsening the degrees of freedom for a Wilson MG lattice at constant lattice geometry.  It's not something we presently care about, but I don't want to hard code against this possibility with the below logic.

I think a more robust enablement of KD optimized is required here.

Reply:
One approach would be flipping the logic, checking if `(geoBlockSize != 1 && transfer_type == QUDA_TRANSFER_OPTIMIZED_KD)`. Thoughts @maddyscientist ?

Reply:
That should work I think.

Reply:
Ah, well, this'll work in the other case you tagged. This routine doesn't know about the transfer_type. We can make this check more robust though: if the geometric block size is 1 and the `ColorSpinorField` invoking `CreateCoarse` is `nSpin == 1` and the input `nVec == nColor` (the convention for the optimized KD op), then we know we're creating something that preserves volume and dof. If anything that motivates replacing `is_optimized_kd` with `is_mg_copy` or something.

Reply:
For now we could go with `is_staggered_layout_preserving` in lieu of thinking through just using `ColorSpinorParam` copies, etc.

Reply:
Or the current name, just with more comments...